### PR TITLE
fix token_valid_mask mismatch for function signature #838

### LINKED ISSUE #838: [Bug] when using fused moe, qwen3_moe funciton "token_valid_mask" signature missmatch.
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [ ] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [ ] 5. Please use English, otherwise it will be closed.

### Describe the bug

the qwen3_moe with: Image, but in moe layer:

Image this will meet "token_valid_mask" miss match  

### Reproduction

MODEL=/models/Qwen3-30B-A3B 
PORT=30161
python -u -m sgl_jax.launch_server \
  --model-path "$MODEL" \
  --tp-size 4 --ep-size 4 --moe-backend fused \
  --enable-single-process \
  --host 127.0.0.1 --port "$PORT" \
  --trust-remote-code \
  --context-length 2048 \
  --max-total-tokens 4096 \
  --max-prefill-tokens 512 \
  --mem-fraction-static 0.3 \
  --disable-precompile --skip-server-warmup \
  --log-level info > "$LOG" 2>&1 &

### Environment

Python: 3.12.12 (main, Feb 12 2026, 00:42:14) [Clang 21.1.4 ]
sglang-jax: 0.0.2
jax: 0.8.1
jaxlib: 0.8.1
triton: Module Not Found
transformers: 4.57.6
numpy: 2.2.6
aiohttp: 3.13.3
fastapi: 0.116.2
huggingface_hub: 0.34.6
modelscope: 1.28.2
orjson: 3.11.7
packaging: 26.0
psutil: 7.0.0
pydantic: 2.11.10
python-multipart: 0.0.22
pyzmq: 27.0.2
uvicorn: 0.35.0
uvloop: 0.21.0
openai: 2.24.0
tiktoken: 0.12.0
[TPU v5-0]: TPU_0(process=0,(0,0,0,0))
[TPU v5-1]: TPU_1(process=0,(1,0,0,0))
[TPU v5-2]: TPU_2(process=0,(0,1,0,0))
[TPU v5-3]: TPU_3(process=0,(1,1,0,0))
ulimit soft: 1048576

### PR DESCRIPTION
**Pull request by cjx0709**
## Motivation

This PR fixes the `token_valid_mask` mismatch in the function signature.
**Closes #838**

## Checklist

- [ ] Please use English, otherwise it will be closed.
- [ ] The purpose of the PR, or link existing issues this PR will resolve.
- [ ] The test plan, such as providing test command.
- [ ] (Optional) The necessary documentation update.

### CODE REVIEW

**Review by JamesBrianD — APPROVED** [review]
LGTM
